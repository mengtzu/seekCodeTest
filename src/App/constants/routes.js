const root = '/';
const adPacks = '/adPacks';
const checkout = '/checkout';
const confirmation = '/confirmation';
const logout = '/Login/Logout';

export const routes = {
    root,
    adPacks,
    checkout,
    confirmation,
    logout
};

export default routes;
