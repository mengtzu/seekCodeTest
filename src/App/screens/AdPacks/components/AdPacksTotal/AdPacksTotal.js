import React from 'react';
import { connect } from 'react-redux';
import {
    Text,
    PageBlock,
    Section,
    Button
} from 'seek-style-guide/react';
import { formatMoney } from 'accounting-js';
import styles from './AdPacksTotal.less';

import { calculateTotal } from '../../../../../shared/calculatePrice/calculatePrice';
import PlusGST from '../PlusGST/PlusGST';

const mapStateToProps = (state) => {
    return {
        products: state.products,
        discounts: state.advertiser.discounts,
        adPacksForm: state.form.adPacksForm
    };
};

const renderSavings = (savings) => {
    if (savings) {
        return (
            <span className={styles.savings}>(You save {formatMoney(savings)}!)</span>
        )
    } else {
        return null;
    }
};

export const AdPacksTotal = ({ products, discounts, adPacksForm, submitting}) => {
    const calculatedTotal = calculateTotal({
        discounts,
        products,
        quantities: adPacksForm.values
    });

    if (calculatedTotal.subTotal) {
        return (
            <PageBlock>
                <Section>
                    <Text heading>Total</Text>
                    <Text subheading>{formatMoney(calculatedTotal.subTotal)} <PlusGST /> {renderSavings(calculatedTotal.savings)}</Text>

                    <Button type="submit" loading={submitting} color="pink" className={styles.checkoutButton}>Checkout</Button>
                </Section>
            </PageBlock>
        )
    } else {
        return null;
    }
};

export default connect(mapStateToProps)(AdPacksTotal);
