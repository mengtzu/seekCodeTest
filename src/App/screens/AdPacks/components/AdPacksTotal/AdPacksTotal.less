@import (reference) "~seek-style-guide/theme";

.checkoutButton {
    margin-top: @grid-row-height * 2;
}

.savings {
    color: @sk-green;

    @media only screen and (min-width: @responsive-breakpoint) {
        display: block;
    }
}