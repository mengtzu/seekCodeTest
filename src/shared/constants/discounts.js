export const DISCOUNT_TYPE_FLAT_RATE = 'flatRate';
export const DISCOUNT_TYPE_THRESHOLD = 'threshold';
export const DISCOUNT_TYPE_X_FOR_Y = 'xfory';
