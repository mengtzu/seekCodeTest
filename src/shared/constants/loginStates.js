export const AUTH_STATUS_UNAUTHENTICATED = 'pending'; //'unauthenticated' exposes the real SEEK login, don't want that!
export const AUTH_STATUS_AUTHENTICATED = 'authenticated';
