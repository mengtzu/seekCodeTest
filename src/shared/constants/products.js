export const CLASSIC = 'classic';
export const STANDOUT = 'standout';
export const PREMIUM = 'premium';
