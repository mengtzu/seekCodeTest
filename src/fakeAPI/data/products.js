import { CLASSIC, STANDOUT, PREMIUM } from '../../shared/constants/products';

let productsData = {};

//Bracket notation isn't necessarily preferred but here it lets us avoid having to iterate over this object in various places

productsData[CLASSIC] = {
    id: CLASSIC,
    basePrice: 269.99
};

productsData[STANDOUT] = {
    id: STANDOUT,
    basePrice: 322.99
};

productsData[PREMIUM] = {
    id: PREMIUM,
    basePrice: 394.99
}

export default productsData;