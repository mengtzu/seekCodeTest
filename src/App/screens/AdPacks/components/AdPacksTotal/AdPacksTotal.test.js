import React from 'react';
import { AdPacksTotal } from './AdPacksTotal';
import renderer from 'react-test-renderer';

import { DISCOUNT_TYPE_FLAT_RATE, DISCOUNT_TYPE_X_FOR_Y, DISCOUNT_TYPE_THRESHOLD } from '../../../../../shared/constants/discounts';
import { CLASSIC, STANDOUT, PREMIUM } from '../../../../../shared/constants/products';

const mockDiscounts = [{
    type: DISCOUNT_TYPE_FLAT_RATE,
    ad: CLASSIC,
    price: 199.99
}, {
    type: DISCOUNT_TYPE_THRESHOLD,
    ad: CLASSIC,
    price: 22,
    threshold: 4
}, {
    type: DISCOUNT_TYPE_X_FOR_Y,
    ad: CLASSIC,
    x: 5,
    y: 1
}];

const mockProducts = {};
mockProducts[CLASSIC] = { basePrice: 777 };
mockProducts[STANDOUT] = { basePrice: 7777 };
mockProducts[PREMIUM] = { basePrice: 77777 };

it('renders correctly with valid discounts', () => {
    const cart = {
        values: {}
    };
    cart.values[CLASSIC] = { quantity: 22 };
    cart.values[STANDOUT] = { quantity: 2 };
    cart.values[PREMIUM] = { quantity: 9 };

    const tree = renderer.create(
        <AdPacksTotal products={mockProducts} discounts={mockDiscounts} adPacksForm={cart} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with no valid discounts', () => {
    const cart = {
        values: {}
    };
    cart.values[CLASSIC] = { quantity: 22 };
    cart.values[STANDOUT] = { quantity: 2 };
    cart.values[PREMIUM] = { quantity: 9 };

    const tree = renderer.create(
        <AdPacksTotal products={mockProducts} discounts={[]} adPacksForm={cart} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with nothing in cart', () => {
    const cart = {
        values: {}
    };
    cart.values[CLASSIC] = { quantity: 0 };
    cart.values[STANDOUT] = { quantity: 0 };
    cart.values[PREMIUM] = { quantity: 0 };

    const tree = renderer.create(
        <AdPacksTotal products={mockProducts} discounts={mockDiscounts} adPacksForm={cart} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});