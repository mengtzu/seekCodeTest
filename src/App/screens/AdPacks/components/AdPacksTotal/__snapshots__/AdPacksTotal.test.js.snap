// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no valid discounts 1`] = `
<div
  className=""
>
  <div
    className="content"
  >
    <div
      className="root"
    >
      <div
        className="root heading"
      >
        <span
          className=""
        >
          Total
        </span>
      </div>
      <div
        className="root subheading"
      >
        <span
          className=""
        >
          $732,641.00
           
          <span
            className="gstLabel"
          >
            +GST
          </span>
           
        </span>
      </div>
      <button
        className="root checkoutButton root_isPink"
        disabled={false}
        type="submit"
      >
        Checkout
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly with nothing in cart 1`] = `null`;

exports[`renders correctly with valid discounts 1`] = `
<div
  className=""
>
  <div
    className="content"
  >
    <div
      className="root"
    >
      <div
        className="root heading"
      >
        <span
          className=""
        >
          Total
        </span>
      </div>
      <div
        className="root subheading"
      >
        <span
          className=""
        >
          $715,679.00
           
          <span
            className="gstLabel"
          >
            +GST
          </span>
           
          <span
            className="savings"
          >
            (You save 
            $16,962.00
            !)
          </span>
        </span>
      </div>
      <button
        className="root checkoutButton root_isPink"
        disabled={false}
        type="submit"
      >
        Checkout
      </button>
    </div>
  </div>
</div>
`;
